spec/associations/active_record_extensions_spec.rb

```diff
@@ -3,65 +3,115 @@
         connection.create_table(:countries, :force => true) do |t|
           t.string :name
         end
         extend ActiveHash::Associations::ActiveRecordExtensions
       end
 
-      class School < ActiveRecord::Base
+      define_ephemeral_class(:School, ActiveRecord::Base) do
         establish_connection :adapter => "sqlite3", :database => ":memory:"
         connection.create_table(:schools, :force => true) do |t|
           t.integer :country_id
           t.string :locateable_type
           t.integer :locateable_id
           t.integer :city_id
         end
+
         extend ActiveHash::Associations::ActiveRecordExtensions
       end
 
-      class City < ActiveHash::Base
+      define_ephemeral_class(:City, ActiveHash::Base) do
         include ActiveHash::Associations
       end
 
-      class Author < ActiveHash::Base
+      define_ephemeral_class(:SchoolStatus, ActiveHash::Base)
+    end
+
+    def define_doctor_classes
+      define_ephemeral_class(:Physician, ActiveHash::Base) do
         include ActiveHash::Associations
-      end
 
-      class SchoolStatus < ActiveHash::Base
+        has_many :appointments
+        has_many :patients, through: :appointments
+
+        self.data = [
+          {:id => 1, :name => "ikeda"},
+          {:id => 2, :name => "sato"}
+        ]
       end
 
-      class Book < ActiveRecord::Base
+      define_ephemeral_class(:Appointment, ActiveRecord::Base) do
         establish_connection :adapter => "sqlite3", :database => ":memory:"
-        connection.create_table(:books, :force => true) do |t|
-          t.integer :author_id
-          t.integer :author_code
-          t.boolean :published
+        connection.create_table :appointments, force: true do |t|
+          t.references :physician
+          t.references :patient
         end
 
-        if Object.const_defined?(:ActiveModel)
-          scope(:published, proc { where(:published => true) })
-        else
-          named_scope :published, {:conditions => {:published => true}}
+        extend ActiveHash::Associations::ActiveRecordExtensions
+
+        belongs_to :physician
+        belongs_to :patient
+      end
+
+      define_ephemeral_class(:Patient, ActiveRecord::Base) do
+        establish_connection :adapter => "sqlite3", :database => ":memory:"
+        connection.create_table :patients, force: true do |t|
         end
+
+        extend ActiveHash::Associations::ActiveRecordExtensions
+
+        has_many :appointments
+        has_many :physicians, through: :appointments
       end
+
+    end
+
+    before do
+      @ephemeral_classes = []
     end
 
     after do
-      Object.send :remove_const, :City
-      Object.send :remove_const, :Author
-      Object.send :remove_const, :Country
-      Object.send :remove_const, :School
-      Object.send :remove_const, :SchoolStatus
-      Object.send :remove_const, :Book
+      @ephemeral_classes.each do |klass_name|
+        Object.send :remove_const, klass_name
+      end
```

Comment:
Do we need to do more here?

On another project I needed to tell rails 6.1 to forget about an active record model that was cached. This crept up when using STI. Maybe we do not need to do this since we are not using STI here?

snipped I had to use, but again, this was a different use case and this code may be unnecessary.

```ruby
  # these descendants are stored in a cache in active support
  # this cleans out those values so future runs do not have bogus classes
  # this causes sporadic test failures.
  def cleanup_subclass(parent, children)
    tracker = ActiveSupport::DescendantsTracker.class_variable_get(:@@direct_descendants)[parent]
    tracker&.reject! { |child| children.include?(child) }
  end
```

Reply:
Yea, and my code referenced an internal rails private variable. Nothing could go wrong with that :(

I really like this change though.
I suppose it improves the original and does not make it worse